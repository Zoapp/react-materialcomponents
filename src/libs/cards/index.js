/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
