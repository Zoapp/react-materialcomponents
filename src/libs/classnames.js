/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
