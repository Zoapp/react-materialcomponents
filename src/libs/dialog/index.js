/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
