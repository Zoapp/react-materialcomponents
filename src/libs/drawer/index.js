import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
