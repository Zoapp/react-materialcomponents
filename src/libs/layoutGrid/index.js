/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
