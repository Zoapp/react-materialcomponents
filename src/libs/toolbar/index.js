/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Toolbar from "./toolbar";
import ToolbarRow from "./row";
import ToolbarSection from "./section";
import ToolbarTitle from "./title";
import ToolbarIcon from "./icon";

export { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon };
