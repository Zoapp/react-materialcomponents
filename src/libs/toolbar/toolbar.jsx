/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * See:
 * https://material.io/components/web/catalog/toolbar/
 * https://material-components-web.appspot.com/toolbar/index.html
 *
 * TODO:
 * - waterfall, flexible, fixedlastrow
 * See:
 * https://github.com/material-components/material-components-web/blob/master/packages/mdc-toolbar/foundation.js
 *
 */

const MDC_TOOLBAR = "mdc-toolbar";

export default class Toolbar extends PureComponent {
  render() {
    const {
      children, fixed, ...otherProps
    } = this.props;
    let classes = MDC_TOOLBAR;
    if (fixed) {
      classes += " mdc-toolbar--fixed";
    }
    const element = (
      <header className={classes} >{children}</header>
    );
    return Rmdc.render(element, otherProps);
  }
}

Toolbar.defaultProps = {
  mdcElement: MDC_TOOLBAR,
  children: null,
  fixed: false,
};

Toolbar.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  fixed: PropTypes.bool,
};
