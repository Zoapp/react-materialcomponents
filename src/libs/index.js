/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
