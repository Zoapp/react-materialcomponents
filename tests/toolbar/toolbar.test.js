import React from "react";
import renderer from "react-test-renderer";
import { Toolbar } from "libs/toolbar";

describe("toolbar/Toolbar", () => {
  it("can render children", () => {
    const component = renderer.create(<Toolbar>Children are here!</Toolbar>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
