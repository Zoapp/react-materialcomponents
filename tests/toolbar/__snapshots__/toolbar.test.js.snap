// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toolbar/Toolbar can render children 1`] = `
<header
  className="mdc-toolbar"
>
  Children are here!
</header>
`;
